### PR TITLE
Added support for SuperMini NRF52840

### PR DESCRIPTION
**Pull request by Axeia**
I did create an 'issue' for this ( see https://github.com/adafruit/circuitpython/issues/8473 ) as I wasn't sure if it warranted its own build. 
The SuperMini NRF52840 is fully compatible with the nice!nano and actually works fine just flashing the nice!nano circuitpython build to this board. 

These files are thus near identical to the nice!nano just changing the USB PID to the one I got from the board whilst it wasn't flashed with anything yet through `lsusb` command under linux ( Bus 001 Device 003: ID 239a:8029 ) and the board/company name where relevant. Once this has its own circuitpython.org/board/supermini_nrf52840 page I'd be happy to update that with a link to the manufacturers wiki etc.

### CODE REVIEW

**Comment by tannewt** [comment]
0x239a is Adafruit's USB vendor ID and shouldn't be used for this board. I'd suggest https://github.com/pidcodes/pidcodes.github.com for requesting these boards you have. I approve for third-party, open source use on closed source boards. Make sure to link to your PR here as proof of being open source software.

File: ports/nrf/boards/supermini_nrf52840/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,8 @@
+USB_VID = 0x239A
+USB_PID = 0x8029
```
Comment (dhalbert) [inline ports/nrf/boards/supermini_nrf52840/mpconfigboard.mk:2]: Awaiting update to approved pid.codes VID/PID.

**Review by dhalbert — CHANGES_REQUESTED** [review]

File: ports/nrf/boards/supermini_nrf52840/mpconfigboard.mk (hunk 1 above)
Comment (tannewt) [reply ports/nrf/boards/supermini_nrf52840/mpconfigboard.mk:2]: pid.codes PR is merged. Please update the values here.

Comment (Axeia) [reply ports/nrf/boards/supermini_nrf52840/mpconfigboard.mk]: requested changes implemented.

**Review by dhalbert — APPROVED** [review]
:+1:

**Pull request merged**